test/TesterInternal/StreamingTests/SingleStreamTestRunner.cs

```diff
@@ -372,6 +374,54 @@ private void Heading(string testName)
             await StopProxies();
         }
 
+        public async Task StreamTest_22_TestImmutabilityDuringStreaming()
+        {
+            Heading("StreamTest_22_TestImmutabilityDuringStreaming");
+
+            IStreamingImmutabilityTestGrain itemProducer = GrainClient.GrainFactory.GetGrain<IStreamingImmutabilityTestGrain>(Guid.NewGuid());
+            string producerSilo = await itemProducer.GetSiloIdentifier();
+
+            // Setup consumers one in silo of item producer and the other one in another silo.
+            IStreamingImmutabilityTestGrain consumerSameSilo = null;
+            IStreamingImmutabilityTestGrain consumerDifferentSilo = null;
+            do
+            {
+                var itemConsumer = GrainClient.GrainFactory.GetGrain<IStreamingImmutabilityTestGrain>(Guid.NewGuid());
+                var consumerSilo = await itemConsumer.GetSiloIdentifier();
+
+                if (consumerSilo != producerSilo && consumerDifferentSilo == null)
+                    consumerDifferentSilo = itemConsumer;
+                if (consumerSilo == producerSilo && consumerSameSilo == null)
+                    consumerSameSilo = itemConsumer;
+            } while (consumerSameSilo == null || consumerDifferentSilo == null);
+
+            // Test behavior if immutability is disabled
+            await consumerSameSilo.SubscribeToStream(itemProducer.GetPrimaryKey(), SMS_STREAM_PROVIDER_NAME_NO_IMMUTABILITY);
+            await consumerDifferentSilo.SubscribeToStream(itemProducer.GetPrimaryKey(), SMS_STREAM_PROVIDER_NAME_NO_IMMUTABILITY);
+
+            await itemProducer.ProduceTestObject(SMS_STREAM_PROVIDER_NAME_NO_IMMUTABILITY);
+
+            Assert.IsFalse(await consumerSameSilo.TestObjectWasTransferredImmutable());
+            Assert.IsFalse(await consumerDifferentSilo.TestObjectWasTransferredImmutable());
+
+            await consumerSameSilo.UnsubscribeFromStream();
+            await consumerDifferentSilo.UnsubscribeFromStream();
+
+            // Test behavior if immutability is enabled
+            itemProducer = GrainClient.GrainFactory.GetGrain<IStreamingImmutabilityTestGrain>(Guid.NewGuid());
```

Comment:
Usage of `Immutable` stays disabled if I reuse the existing itemProducer. Any idea why this happens or is this intended behavior?